### ISSUE TITLE
random disappreance of ListView Items with FluidControls.ListItem  on fast scrolling

### ISSUE THREAD

**Issue by rmallah** [lirios/fluid#115]
It has been observed that ListViews using base delegate `FluidControls.ListItem` 
may fail to instantiate on fast scrolling and the following error comes in the Application 
output log window:

```
<Unknown File>: QML VisualDataModel: Error creating delegate
<Unknown File>: QML VisualDataModel: Error creating delegate
<Unknown File>: QML VisualDataModel: Error creating delegate
<Unknown File>: QML VisualDataModel: Error creating delegate
<Unknown File>: QML VisualDataModel: Error creating delegate
<Unknown File>: QML VisualDataModel: Error creating delegate
```

The functional impact of above error has only been observed in mobile
platform ( videos links shall be attached) . 

A comparison has been made by using a simple delegate in place of 
`FluidControls.ListItem`  and the log messages and problem disappears.
The selection of kind of delegate can be made by toggling the `simple`
property in code below: 

the minimal code that can be used to demonstrate it is

```
import QtQuick 2.0


import QtQuick 2.6
import QtQuick.Controls 2.1
import QtQuick.Controls.Material 2.1
import QtQuick.Layouts 1.1

import Fluid.Controls 1.0 as FluidControls

import "../components"


FluidControls.Page {

    property  int simple : 0

    id: root


    ListView {
        id: listView
        anchors.fill: parent
        anchors.leftMargin: 10
        anchors.rightMargin: 10
        model: mymodel
        delegate: simple ? simpledelegate : fluidDelegate
    }

    Component {
        id: fluidDelegate
        FluidControls.ListItem {
            iconName : "action/assignment"
            leftItem : Text { text: "left item" }
            maximumLineCount: 3
            rightItem : Text { text: "right item" }
            secondaryItem : Text { text: "secondary item" }
            subText: message
            valueText: '('+sno+') ' + subject
            showDivider: true
        }
    }

    Component {
        id: simpledelegate
        Item {
            height: (label1.implicitHeight + label2.implicitHeight + rect1.implicitHeight) * 1.5
            ColumnLayout {
                spacing: 5
                Label {
                    id: label1
                    text: '('+sno+') ' + subject
                    font.pixelSize: 14
                    width: listView.width
                }
                Text {
                    id: label2
                    text: message
                    width: listView.width
                    font.pixelSize: 10
                }
                Rectangle {
                    id: rect1
                    color:"#999"
                    width: listView.width
                    height: 1
                }
            }
        }
    }


    ListModel {
        id: mymodel
        ListElement {
            sno: 1
            subject:"hello"
            message:"there"
        }
    }

    Component.onCompleted: {
        mymodel.clear();
        for(var i=1 ; i <= 50 ; i++) {

             //  use simple string if Lorem is unavailable.
             // and comment out import of ../components

             var subject = Lorem.loremIpsumSentence(5) ;
             var message = Lorem.loremIpsumParagraph(20);

            //var subject = "subject line " + i  ;
            //var message = "message text " + i  ;

            mymodel.append( { "sno": i , "subject": subject , "message": message  }  );
        }
    }
}
```

Videos:

Problem case:  https://drive.google.com/open?id=0B4yfAVMLmbQxQWZTZFdPZ0NEc3M
Simple case:     https://drive.google.com/open?id=0B4yfAVMLmbQxQmV5QUlOMkJFQWc

**Comment by rmallah** [comment]
the delegates are numbered in first video , so that missing ones can be observed.

**Comment by rmallah** [comment]
Related: https://bugreports.qt.io/browse/QTBUG-49224

**Comment by rmallah** [comment]
It is being hypothesized that omission of background containing the Ripple and ThinDivider helps
in minimizing the behaviour.  supporting details shall be presented .

**Comment by plfiorini** [comment]
@rmallah Do you still have this problem with current `develop` branch?

The ListItem is reusing the ItemDelegate background, this means it doesn't override it anymore.

**Comment by redgrapetech** [comment]
Hi Pier,
pls allow a few days. I am in middle of something else.
I will check with the current develop branch.

>> "The ListItem is reusing the ItemDelegate background, this means it doesn't override it anymore."
could not understand above much at this moment. I will have a re-look .

regds
mallah.

**Comment by plfiorini** [comment]
Sure @redgrapetech don't worry

**Comment by plfiorini** [comment]
Sounds like this bug: https://bugreports.qt.io/browse/QTBUG-50992?filter=18322

**Comment by rmallah** [comment]
@plfiorini  I upvoted the upstream bug today

**Comment by Talkless** [comment]
Try this workaround: https://bugreports.qt.io/browse/QTBUG-50992?focusedCommentId=379090&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-379090

**Comment by plfiorini** [comment]
Upstream bug is fixed in Qt 5.9.4 and 5.10.1
